<img src="https://s2.loli.net/2022/07/04/rD5a9fk21iVJ7KG.jpg" alt="drawing" width="450"/>

# rosalind-python

Solution in Python for problems in [ROSALIND](https://rosalind.info/)


## Bioinfomatics Stronghold


| No. | Title | URL | Script |
| :----- | :----- | :---- | :---- |
| 001 | Counting DNA Nucleotides | [dna](https://rosalind.info/problems/dna/) | [001.dna.py](./bioinfomatics-stronghold/001.dna.py) |
| 002 | Transcribing DNA into RNA | [rna](https://rosalind.info/problems/rna/) | [002.rna.py](./bioinfomatics-stronghold/002.rna.py) |
| 003 | Complementing a Strand of DNA | [revc](https://rosalind.info/problems/revc/) | [003.revc.py](./bioinfomatics-stronghold/003.revc.py) |
| 004 | Rabbits and Recurrence Relations | [fib](https://rosalind.info/problems/fib/) | [004.fib.py](./bioinfomatics-stronghold/004.fib.py) |
| 005 | Computing GC Content | [gc](https://rosalind.info/problems/gc/) | [005.gc.py](./bioinfomatics-stronghold/005.gc.py) |
| 006 | Counting Point Mutations | [hamm](https://rosalind.info/problems/hamm/) | [006.hamm.py](./bioinfomatics-stronghold/006.hamm.py) |
| 007 | Mendel&#39;s First Law | [iprb](https://rosalind.info/problems/iprb/) | [007.iprb.py](./bioinfomatics-stronghold/007.iprb.py) |
| 008 | Translating RNA into Protein | [prot](https://rosalind.info/problems/prot/) | [008.prot.py](./bioinfomatics-stronghold/008.prot.py) |
| 009 | Finding a Motif in DNA | [subs](https://rosalind.info/problems/subs/) | [009.subs.py](./bioinfomatics-stronghold/009.subs.py) |
| 010 | Consensus and Profile | [cons](https://rosalind.info/problems/cons/) | [010.cons.py](./bioinfomatics-stronghold/010.cons.py) |
| 011 | Mortal Fibonacci Rabbits | [fibd](https://rosalind.info/problems/fibd/) | [011.fibd.py](./bioinfomatics-stronghold/011.fibd.py) |
| 012 | Overlap Graphs | [grph](https://rosalind.info/problems/grph/) | [012.grph.py](./bioinfomatics-stronghold/012.grph.py) |
| 013 | Calculating Expected Offspring | [iev](https://rosalind.info/problems/iev/) | [013.iev.py](./bioinfomatics-stronghold/013.iev.py) |
| 014 | Finding a Shared Motif | [lcsm](https://rosalind.info/problems/lcsm/) | [014.lcsm.py](./bioinfomatics-stronghold/014.lcsm.py) |
| 015 | Independent Alleles | [lia](https://rosalind.info/problems/lia/) | [015.lia.py](./bioinfomatics-stronghold/015.lia.py) |
| 016 | Finding a Protein Motif | [mprt](https://rosalind.info/problems/mprt/) | [016.mprt.py](./bioinfomatics-stronghold/016.mprt.py) |
| 017 | Inferring mRNA from Protein | [mrna](https://rosalind.info/problems/mrna/) | [017.mrna.py](./bioinfomatics-stronghold/017.mrna.py) |
| 018 | Open Reading Frames | [orf](https://rosalind.info/problems/orf/) | [018.orf.py](./bioinfomatics-stronghold/018.orf.py) |
| 019 | Enumerating Gene Orders | [perm](https://rosalind.info/problems/perm/) | [019.perm.py](./bioinfomatics-stronghold/019.perm.py) |
| 020 | Calculating Protein Mass | [prtm](https://rosalind.info/problems/prtm/) | [020.prtm.py](./bioinfomatics-stronghold/020.prtm.py) |
| 021 | Locating Restriction Sites | [revp](https://rosalind.info/problems/revp/) | [021.revp.py](./bioinfomatics-stronghold/021.revp.py) |
| 022 | RNA Splicing | [splc](https://rosalind.info/problems/splc/) | [022.splc.py](./bioinfomatics-stronghold/022.splc.py) |
| 023 | Enumerating k-mers Lexicographically | [lexf](https://rosalind.info/problems/lexf/) | [023.lexf.py](./bioinfomatics-stronghold/023.lexf.py) |
| 024 | Longest Increasing Subsequence | [lgis](https://rosalind.info/problems/lgis/) | [024.lgis.py](./bioinfomatics-stronghold/024.lgis.py) |
| 025 | Genome Assembly as Shortest Superstring | [long](https://rosalind.info/problems/long/) | [025.long.py](./bioinfomatics-stronghold/025.long.py) |
| 026 | Perfect Matchings and RNA Secondary Structures | [pmch](https://rosalind.info/problems/pmch/) | [026.pmch.py](./bioinfomatics-stronghold/026.pmch.py) |
| 027 | Partial Permutations | [pper](https://rosalind.info/problems/pper/) | [027.pper.py](./bioinfomatics-stronghold/027.pper.py) |
| 028 | Introduction to Random Strings | [prob](https://rosalind.info/problems/prob/) | [028.prob.py](./bioinfomatics-stronghold/028.prob.py) |
| 029 | Enumerating Oriented Gene Orderings | [sign](https://rosalind.info/problems/sign/) | [029.sign.py](./bioinfomatics-stronghold/029.sign.py) |
| 030 | Finding a Spliced Motif | [sseq](https://rosalind.info/problems/sseq/) | [030.sseq.py](./bioinfomatics-stronghold/030.sseq.py) |
| 031 | Transitions and Transversions | [tran](https://rosalind.info/problems/tran/) | [031.tran.py](./bioinfomatics-stronghold/031.tran.py) |
| 032 | Completing a Tree | [tree](https://rosalind.info/problems/tree/) | [032.tree.py](./bioinfomatics-stronghold/032.tree.py) |
| 033 | Catalan Numbers and RNA Secondary Structures | [cat](https://rosalind.info/problems/cat/) | [033.cat.py](./bioinfomatics-stronghold/033.cat.py) |
| 034 | Error Correction in Reads | [corr](https://rosalind.info/problems/corr/) | [034.corr.py](./bioinfomatics-stronghold/034.corr.py) |
| 035 | Counting Phylogenetic Ancestors | [inod](https://rosalind.info/problems/inod/) | [035.inod.py](./bioinfomatics-stronghold/035.inod.py) |
| 036 | k-Mer Composition | [kmer](https://rosalind.info/problems/kmer/) | [036.kmer.py](./bioinfomatics-stronghold/036.kmer.py) |
| 037 | Speeding Up Motif Finding | [kmp](https://rosalind.info/problems/kmp/) | [037.kmp.py](./bioinfomatics-stronghold/037.kmp.py) |
| 038 | Finding a Shared Spliced Motif | [lcsq](https://rosalind.info/problems/lcsq/) | [038.lcsq.py](./bioinfomatics-stronghold/038.lcsq.py) |
| 039 | Ordering Strings of Varying Length Lexicographically | [lexv](https://rosalind.info/problems/lexv/) | [039.lexv.py](./bioinfomatics-stronghold/039.lexv.py) |
| 040 | Maximum Matchings and RNA Secondary Structures | [mmch](https://rosalind.info/problems/mmch/) | [040.mmch.py](./bioinfomatics-stronghold/040.mmch.py) |
| 041 | Creating a Distance Matrix | [pdst](https://rosalind.info/problems/pdst/) | [041.pdst.py](./bioinfomatics-stronghold/041.pdst.py) |
| 043 | Matching Random Motifs | [rstr](https://rosalind.info/problems/rstr/) | [043.rstr.py](./bioinfomatics-stronghold/043.rstr.py) |
| 044 | Counting Subsets | [sset](https://rosalind.info/problems/sset/) | [044.sset.py](./bioinfomatics-stronghold/044.sset.py) |
| 045 | Introduction to Alternative Splicing | [aspc](https://rosalind.info/problems/aspc/) | [045.aspc.py](./bioinfomatics-stronghold/045.aspc.py) |
| 046 | Edit Distance | [edit](https://rosalind.info/problems/edit/) | [046.edit.py](./bioinfomatics-stronghold/046.edit.py) |

## Bioinfomatics Armory


| No. | Title | URL | Script |
| :----- | :----- | :---- | :---- |
| 001 | Introduction to the Bioinformatics Armory | [ini](https://rosalind.info/problems/ini/) | [001.ini.py](./bioinfomatics-armory/001.ini.py) |

## Bioinfomatics Textbook Track


| No. | Title | URL | Script |
| :----- | :----- | :---- | :---- |
| 001 | Compute the Probability of a Hidden Path | [ba10a](https://rosalind.info/problems/ba10a/) | [001.ba10a.py](./bioinfomatics-textbook-track/001.ba10a.py) |

## Algorithmic Heights


| No. | Title | URL | Script |
| :----- | :----- | :---- | :---- |
| 001 | Fibonacci Numbers | [fibo](https://rosalind.info/problems/fibo/) | [001.fibo.py](./algorithmic-heights/001.fibo.py) |
| 002 | Binary Search | [bins](https://rosalind.info/problems/bins/) | [002.bins.py](./algorithmic-heights/002.bins.py) |
| 003 | Degree Array | [deg](https://rosalind.info/problems/deg/) | [003.deg.py](./algorithmic-heights/003.deg.py) |
| 004 | Insertion Sort | [ins](https://rosalind.info/problems/ins/) | [004.ins.py](./algorithmic-heights/004.ins.py) |
| 005 | Double-Degree Array | [ddeg](https://rosalind.info/problems/ddeg/) | [005.ddeg.py](./algorithmic-heights/005.ddeg.py) |
| 006 | Majority Element | [maj](https://rosalind.info/problems/maj/) | [006.maj.py](./algorithmic-heights/006.maj.py) |
